File: leetcode/701-insert-into-a-binary-search-tree/BSTInsertion.java
public class BSTInsertion {
    private class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        public TreeNode() {}

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    private TreeNode insertIntoBST(TreeNode root, int val) {
        if (root == null)
            return new TreeNode(val);

        if (val < root.val)
            root.left = insertIntoBST(root.left, val);
        else
            root.right = insertIntoBST(root.right, val);
        
        return root;
    }
}

File: leetcode/701-insert-into-a-binary-search-tree/bst_insertion.py
# LC 701: Insert into a binary search tree
class Solution:
    def bst_insertion(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if not root:
            return TreeNode(val)
        
        if val < root.val:
            root.left = self.bst_insertion(root.left, val)
        else:
            root.right = self.bst_insertion(root.right, val)
        
        return root
